spinalcordtoolbox/resampling.py

```diff
@@ -50,7 +50,7 @@ def resample_nib(image, new_size=None, new_size_type=None, image_dest=None, inte
     elif type(image) == Image:
         img = nib.nifti1.Nifti1Image(image.data, image.hdr.get_best_affine())
     else:
-        raise Exception(TypeError)
+        raise TypeError(f'Invalid image type: {type(image)}')
```

Comment:
I'm curious if `try`/`except TypeError` even catches `Exception(TypeError)`?

Bit besides the point, though, since we're replacing it with the proper syntax, anyway. :P

Reply:
Nope: `except TypeError` catches any exception which inherits from TypeError, and the raised exception is an Exception (which happens to contain the class object `TypeError` as one of its fields).
```py3
>>> try: raise Exception(TypeError)
... except TypeError: print('yes')
... else: print('nope')
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception: <class 'TypeError'>
```